src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,15 +123,17 @@ private void checkHealth() {
     while (testIt.hasNext()) {
```

Comment:
At some point the code here was removing dead tests here using the iterator, but since it's not doing that any more this could just use some kind of for(...) loop.  No?